### PR TITLE
Oddball keyboard - trackball improvements

### PR DESCRIPTION
**Pull request by atulloh**
## Description

Improved ADNS9800 trackball code on the Oddball keyboard:
- added scrolling
- added click and drag
- added dynamic CPI and saving CPI setting to EEPROM
- separated keyboard/user code from sensor code
- fixed occasionally erratic motion tracking
- updated ADNS firmware

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

_None_

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by atulloh** [comment]
Bump; this is still good to go. Let me know if there is anything I can do to help with the review.

File: keyboards/oddball/adns/adns.c [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
 #include "quantum.h"
-#include "pointing_device.h"
-#include "adns9800_srom_A4.h"
+#include "adns9800_srom_A6.h"
 #include "../../lib/lufa/LUFA/Drivers/Peripheral/SPI.h"
+#include "adns.h"
```
Comment (drashna) [inline keyboards/oddball/adns/adns.c:20]: I think these can be replaced with spi.h

Comment (fauxpark) [reply keyboards/oddball/adns/adns.c:20]: Yep, this should be using spi_master: https://docs.qmk.fm/#/spi_driver

**Comment by tzarc** [comment]
@atulloh is this just pending a conversion over to the `spi_master` driver mentioned by fauxpark? Were there any further modifications you were going to do?

**Comment by atulloh** [comment]
> @atulloh is this just pending a conversion over to the `spi_master` driver mentioned by fauxpark? Were there any further modifications you were going to do?

Yep, I just need to make the `spi` changes; nothing further.

File: keyboards/oddball/adns/adns.c (hunk 1 above)
Comment (atulloh) [reply keyboards/oddball/adns/adns.c]: Hi, I've now done this, and is working as expected. Just some things I came across:
- I needed to add `spi_master.h` manually to my calling code
- I had to add `src += spi_master.c` to my `rules.mk` to get it to compile, I couldn't find a particular build option anywhere
- I'm not really too familiar with SPI, but I had to `setPinOutput(SPI_SS_PIN)` before calling `spi_init()` to get things working (I saw it did the same in the LUFA SPI_AVR8.h `SPI_Init(...)` function)

**Comment by atulloh** [comment]
Hi, looking at the **PR Lint keyboards** action, which failed, it seems like it was some issue determining file changes rather than linting? Perhaps it was caused by a big merge bringing my branch up to date? Or perhaps I just have lint errors 😛

**Comment by noroadsleft** [comment]
> Hi, looking at the **PR Lint keyboards** action, which failed, it seems like it was some issue determining file changes rather than linting? Perhaps it was caused by a big merge bringing my branch up to date? Or perhaps I just have lint errors stuck_out_tongue

From what I've been able to determine seeing that test fail on other PRs, is that it fails if a file is deleted or moved (not sure about renames).

**Comment by atulloh** [comment]
> > Hi, looking at the **PR Lint keyboards** action, which failed, it seems like it was some issue determining file changes rather than linting? Perhaps it was caused by a big merge bringing my branch up to date? Or perhaps I just have lint errors stuck_out_tongue
> 
> From what I've been able to determine seeing that test fail on other PRs, is that it fails if a file is deleted or moved (not sure about renames).

Any tips on resolving them? Maybe it was my massive merge commit; I'll try a rebase instead.

**Comment by atulloh** [comment]
Looks like it is fixed in an updated version of the `trilom/file-changes-action`: https://github.com/trilom/file-changes-action/issues/104

File: keyboards/oddball/adns/adns.c (hunk 1 above)
Comment (drashna) [reply keyboards/oddball/adns/adns.c]: Yeah, those would all be the correct way to handle things. 

As for the last bit. I think that's a bug, actually. And specifically, somebody called the wrong function:
https://github.com/qmk/qmk_firmware/blob/0f9b7b9768e53add15dc7e860efab0464cd7533d/drivers/avr/spi_master.c#L45-L52

**Comment by drashna** [comment]
> Any tips on resolving them? Maybe it was my massive merge commit; I'll try a rebase instead.

Mostly, "not your problem".  As long as it compiles properly, and according to Travis CI, it does.

**Comment by atulloh** [comment]
I've got a bunch of changes that have superseded this PR, so I'll kill this one off and open another.

**Pull request closed without merging**